### DISCUSSION TITLE
Tuning Data Generating Models

### DISCUSSION THREAD

**Discussion by GForb** [pmsims-package/pmsims#11] [Ideas]
One of the features we currently have is functions that tune the data generating model. Tuning uses a numerical optimizer to find the value of a particular parameter that leads to the desired large sample predictive performance when the analysis model is fit to data from the data generating model. The sample size used for tuning needs to be large enough that model performance has very little sampling variation. 

For example, the strength of association between predictors and outcomes can be tuned to give a desired c-statistic for a logistic regression model

We currently have implemented tuning of a single parameter using the optimize function from the stats package.

**Comment by GForb** [comment]
## Tuning multiple parameters

It is possible to use optim from the stats package to tune multiple parameters. For example we may wish to adjust the model intercept and strength of association between predictors and outcome to give a desired AUC. 

As it is the case that both of these factors may impact sample size, a user may wish to see the sample size for different values.

The downside of tuning two parameters is that it is likely that there will not be a single solution which leads to the desired model performance but instead a (possibly infinite) set of solutions, each of which could lead to a different sample size.

The alternative to multi-parameter tuning would be to force the user to specify the different values they want for one of the parameters (eg. the intercept) and then use single parameter tuning for the other parameter. This could either be implemented as an option in our package or left to users to implement with a loop of some sort.

**Reply by flxzimmer** [reply]
What I came across is the sitatuation where 2 parameters needed tuning (some model parameters) to achieve specific values in 2 criteria (c-stat and prevalence). In this multi-parameter setting there can be one solution again and I think it's not too uncommon. To implement this, in a simulation function one can add the two "deltas" of the criteria and find a minimum in the paramaters using surrogate modeling.

**Comment by flxzimmer** [comment]
### Larger Sample size for lower sampling variation

One issue I see with a large sample size in the data that is used for model fitting for the numerical optimizer is the case when the performance measures increase with sample size (like for accuracy). I'm however not sure how relevant such are for clinical prediction models.

For these, I would suggest to instead use a somewhat realistic sample size in this case. The way it can still work also with `optimize ` is to take a mean over multiple runs (e.g. 1000) for each tested value. If we go more sophisticated (with multiple parameters to be tuned and/or multiple desired performance measures) surrogate modeling could be used instead of optim/optimize and also deal naturally with this noise.

**Reply by GForb** [reply]
Agree that I'm not sure it is relevant at this stage
